### PR TITLE
[orchestrator] collect ResourceRequirements on other workloads as well

### PR DESCRIPTION
**Pull request by nammn**
### What does this PR do?

- bump agent-payload: https://github.com/DataDog/agent-payload/pull/152 <-- also depends on this
- we were missing daemonset tests, added some
- add `ResourceRequirements` collection for the other workload resources (deployment, replicaset, daemonset, statefulset)

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The appropriate `team/..` label has been applied, if known.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: go.mod
Comment (nammn) [inline go.mod:253]: will be updated once this is merged: https://github.com/DataDog/agent-payload/pull/152

File: releasenotes-dca/notes/collect-rr-all-workloads-1334f0c22ff4de04.yaml
Comment (cswatt) [inline releasenotes-dca/notes/collect-rr-all-workloads-1334f0c22ff4de04.yaml:11]:
```suggestion
    Collect ResourceRequirements on other K8s workloads as well for live containers (Deployment, StatefulSet, ReplicaSet, DaemonSet)
```

**Review by cswatt — COMMENTED** [review]
just making one (1) suggestion on the release note—otherwise no docs review required

File: pkg/collector/corechecks/cluster/orchestrator/orchestrator_transformer_test.go
Comment (kangyili) [inline pkg/collector/corechecks/cluster/orchestrator/orchestrator_transformer_test.go:46]: Do you test the template that only have resource limits or only have resource requests ?

Comment (nammn) [reply pkg/collector/corechecks/cluster/orchestrator/orchestrator_transformer_test.go:46]: I currently have one test for this, as we already have those tests in the `podSpec`. But for sure I can add one 👍

**Review by kangyili — APPROVED** [review]
LGTM

File: pkg/collector/corechecks/cluster/orchestrator/orchestrator_transformer_test.go
Comment (nammn) [reply pkg/collector/corechecks/cluster/orchestrator/orchestrator_transformer_test.go:46]: Example tests are here: https://github.com/DataDog/datadog-agent/blob/b39062e76cb14ea02cdb2e1a70770c3c8de2f189/pkg/util/orchestrator/pod_test.go#L450-L463

**Comment by xlucas** [comment]
Could you merge this after the orchestrator check refactoring is merged? 🙏

File: go.mod
Comment (nammn) [inline go.mod:68]: Will be merged by: https://github.com/DataDog/datadog-agent/pull/10722

File: pkg/collector/corechecks/cluster/orchestrator/transformers/k8s/pod.go
Comment (xlucas) [inline pkg/collector/corechecks/cluster/orchestrator/transformers/k8s/pod.go:63]: Suggestion on the method name: to follow the rest of the package I think it would make sense to rename it `ExtractResourceRequirements` (the model part is redundant with the fact we expect to return models from k8s objects in that package and we have named most of our extraction methods `extract<Foo>` or `convert<Foo>` AFAICT).

Also do you mind adding a small comment since that function is exposed ?

File: pkg/collector/corechecks/cluster/orchestrator/transformers/k8s/common_test.go
Comment (xlucas) [inline pkg/collector/corechecks/cluster/orchestrator/transformers/k8s/common_test.go:76]: it's just my feeling but I would have moved to the pod code (since the template is a pod spec) because it makes sense for some resources to depend on the pod transformer (hence depend on its test tooling too)

conversely, not all resources might have a pod spec

File: pkg/collector/corechecks/cluster/orchestrator/transformers/k8s/pod.go
Comment (xlucas) [inline pkg/collector/corechecks/cluster/orchestrator/transformers/k8s/pod.go:78]: why do we need this?

**Review by xlucas — APPROVED** [review]
lgtm overall, couple of suggestions that would be nice to look at before merging 🙏

File: pkg/collector/corechecks/cluster/orchestrator/transformers/k8s/deployment_test.go
Comment (xlucas) [inline pkg/collector/corechecks/cluster/orchestrator/transformers/k8s/deployment_test.go:20]: good catch 👍

File: pkg/collector/corechecks/cluster/orchestrator/transformers/k8s/pod.go
Comment (nammn) [reply pkg/collector/corechecks/cluster/orchestrator/transformers/k8s/pod.go:63]: That makes sense! 
Let me rename it, the problem with passing in `corev1.PodTemplateSpec` is that `pods` are using `PodSpec` while the other workloads are using `PodTemplateSpec`.

So the choices I see are:
- keep using 2 different containerTypes as input
- use 2 different methods
- use an interface and convert when needed

I prefer to keep the first choice, let me know if you think otherwise!

File: pkg/collector/corechecks/cluster/orchestrator/transformers/k8s/common_test.go
Comment (nammn) [reply pkg/collector/corechecks/cluster/orchestrator/transformers/k8s/common_test.go:76]: I moved this to `pod_test.go` let me know if you meant otherwise

File: pkg/collector/corechecks/cluster/orchestrator/transformers/k8s/pod.go
Comment (nammn) [reply pkg/collector/corechecks/cluster/orchestrator/transformers/k8s/pod.go:78]: This is a small hack, because currently the unit tests don't have that field set, meaning they expect the default value, in this case `nil` i.e.
```
model.Deployment.ResourceRequirements = nil
```

So returning explicitly nil was a choice out of laziness, I remembered this pattern being used somewhere else in the code base + I didn't see too many cons/unexpected behaviours in doing so.

Do you prefer to return the empty slice and edit the unit tests?

File: pkg/collector/corechecks/cluster/orchestrator/transformers/k8s/pod.go
Comment (xlucas) [reply pkg/collector/corechecks/cluster/orchestrator/transformers/k8s/pod.go]: the two methods approach is great thanks for updating 👍

**Pull request merged**
